websocket/polygon.go

```diff
@@ -269,8 +262,15 @@ func (c *Client) close(reconnect bool) {
 		if err := c.ptomb.Wait(); err != nil {
 			c.log.Errorf("process thread closed: %v", err)
 		}
+
 		c.shouldClose = true
-		close(c.output)
+
+		select {
+		case <-c.output:
+			c.log.Debugf("output channel was closed by user")
+		default:
+			close(c.output)
+		}
```

Comment:
If a user hasn't read a message off of the output channel would this end up not closing the output channel?

Reply:
i was actually worried about that as well. tried testing that scenario and the channel seemed to close but now i'm skeptical. looking into it a bit more

Reply:
just confirmed that you're right. i added a helper that closes and recovers from the panic 🙂